Bug fix for dump_interactions and continuous_communication file name

davidelct opened the pull request:
Thanks for working on EGG and opening a PR!
Please fill this template to help us understand what are you introducing in EGG.

<!--- Provide a general summary of your changes in the Title above -->

## Description
- Added an `__add__` method to the `Interaction` dataclass
- Changed the return statement in `dump_interactions` function to return the full interaction
- Renamed `continous_communication.py` to `continuous_communication.py` and all the related imports accordingly

## Related Issue (if any)
- The interactions bug could be reproduced by calling the `dump_interactions` function, which would fail raising the TypeError exception: unsupported operand type(s) for +: 'Interaction' and 'Interaction'.

## Motivation and Context
The `Interaction` dataclass lacked an `__add__` method to control the behavior of the + operator between two interaction objects: this would prevent `dump_interactions` from accumulating the interaction objects in the `full_interaction` variable. Moreover, the `dump_interactions` function returned the `interaction` variable instead of the `full_interaction`.
(https://github.com/facebookresearch/EGG/issues/240#issue-1150682221)
The `continuous_communication.py` file name had a typo.
(https://github.com/facebookresearch/EGG/issues/239#issue-1150680172)

## How Has This Been Tested?
Tested by adding two Interaction objects into a single Interaction, and checked that no data is missing. Tested dump_interactions behavior by checking that it returns the full interaction. Pytest from root dir passes all tests.

robertodessi commented:
Hi @davidelct, thanks for opening the PR! It looks great! I left a couple of comments, I think that after integrating those it's good to go! Thanks!

davidelct commented:
hi @robertodessi ! thanks for your suggestions. the following changes have been made to `__add__`:
- doctesting has been added
- aux and aux_info are now added correctly, without overwriting of shared keys
- _check_cat is now used instead of simply using torch.cat()

let me know if there is anything else we could add here and thanks for your help !

robertodessi commented on egg/core/interaction.py:
```diff
@@ -245,6 +310,7 @@ def dump_interactions(
     :param variable_length: whether variable-length communication is used.
     :param device: device (e.g. 'cuda') to be used.
     :return: The entire log of agent interactions, represented as an Interaction instance.
+        Interaction instances.
```
?

robertodessi commented on egg/core/interaction.py:
```diff
@@ -71,6 +71,70 @@ class Interaction:
+                if k in self.aux and k in other.aux:
+                    aux[k] = torch.cat((self.aux[k], other.aux[k]))
+                else:
+                    if k in self.aux:
+                        aux[k] = self.aux[k]
+                    elif k in other.aux:
+                        aux[k] = other.aux[k]
```
```suggestion
                    aux[k] = self.aux[k] if k in self.aux else other.aux[k]
```

robertodessi commented on egg/core/interaction.py:
```diff
@@ -71,6 +71,70 @@ class Interaction:
+                if k in self.aux_input and k in other.aux_input:
+                    aux_input[k] = torch.cat((self.aux_input[k], other.aux_input[k]))
+                else:
+                    if k in self.aux_input:
+                        aux_input[k] = self.aux_input[k]
+                    elif k in other.aux_input:
+                        aux_input[k] = other.aux_input[k]
```
```suggestion
                    aux[k] = self.aux[k] if k in self.aux else other.aux[k]
```

robertodessi commented:
Hi @davidelct,

Thanks a lot for fixing this! It looked already great to me, but I left a minor minor comment (not sure if it's a typo in the docstring) and two suggestions. What do you think?

davidelct commented:
Hi @robertodessi, thanks a lot for the suggestions ! They look good to me, I incorporated as follows:
- in `__add__` the aux dicts are now combined with a single function, so as to avoid repetitions in code and to make it consistent with how the rest of the data is combined too
- yes, there was a typo in the docstrings for `dump_interactions`: now removed
- minor: i made the doctest for `__add__` a bit more readable

thanks !

robertodessi commented:
Thank you!
